lib/Conversion/ImportVerilog/Statement.cpp

```diff
@@ -54,6 +54,55 @@ LogicalResult Context::visitConditionalStmt(
   return success();
 }
 
+LogicalResult Context::createCase(
+    Value caseExpr,
+    nonstd::span<slang::ast::CaseStatement::ItemGroup const> &items,
+    const slang::ast::Statement *defaultCase, unsigned long itemIndex,
+    unsigned long exprIndex, Location loc, Type type) {
+  if (itemIndex >= items.size()) {
+    return convertStatement(defaultCase);
+  }
+  auto itemExpr = convertExpression(*items[itemIndex].expressions[exprIndex]);
+  auto itemStmt = items[itemIndex].stmt;
+  Value cond = builder.create<moore::EqualityOp>(loc, caseExpr, itemExpr);
+  if (!cond.getType().isa<mlir::IntegerType>()) {
+    auto zeroValue = builder.create<moore::ConstantOp>(loc, type, 0);
+    cond = builder.create<moore::InEqualityOp>(loc, cond, zeroValue);
+  }
+
+  auto ifOp = builder.create<mlir::scf::IfOp>(
+      loc, cond, itemIndex < (items.size() - 1) || defaultCase != nullptr);
+  OpBuilder::InsertionGuard guard(builder);
+
+  builder.setInsertionPoint(ifOp.thenYield());
+  if (failed(convertStatement(itemStmt)))
+    return failure();
+
+  if (itemIndex < (items.size() - 1) || defaultCase != nullptr) {
+    builder.setInsertionPoint(ifOp.elseYield());
+    if (exprIndex < (items[itemIndex].expressions.size() - 1)) {
+      return createCase(caseExpr, items, defaultCase, itemIndex, exprIndex + 1,
+                        loc, type);
+    }
+
+    return createCase(caseExpr, items, defaultCase, itemIndex + 1, 0, loc,
+                      type);
+  }
+  return success();
```

Comment:
Similar to the `inside` expression above: can you make this non-recursive and instead iterate over the `items` and build up a sequence of `ifOp`s?